spec/models/ab_experiment/goal_conversion_handler_spec.rb

```diff
@@ -0,0 +1,151 @@
+require "rails_helper"
+RSpec.describe AbExperiment::GoalConversionHandler do
+  include FieldTest::Helpers
+
+  describe ".call" do
+    subject(:handler) { described_class.new(user: user, goal: goal, experiments: experiments) }
+
+    let(:user) { create(:user) }
+    let(:goal) { "non_sense" }
+    let(:experiments) { FieldTest.config["experiments"] }
+
+    context "with no experiments" do
+      let(:experiments) { nil }
+
+      it "gracefully handles a case where there are no tests" do
+        handler.call
+        expect(FieldTest::Event.all.size).to be(0)
+      end
+    end
+
+    context "with experiment that started to soon for some results" do
+      let(:goal) { described_class::USER_CREATES_COMMENT_GOAL }
+      let(:experiment_name) { AbExperiment::CURRENT_FEED_STRATEGY_EXPERIMENT }
+      # NOTE: We're choosing a future date as a logic short-cut for comment tests
+      let(:experiments) { { experiment_name => { "start_date" => 8.days.from_now } } }
+
+      before do
+        field_test(experiment_name, participant: user)
+      end
+
+      it "registers some events but not others" do
```

Comment:
Quick, non-blocking suggestion: let's aggregate the failures here since we have multiple expectations:
```suggestion
      it "registers some events but not others", :aggregate_failures do
```